### PR TITLE
Example stable plugin

### PR DESCRIPTION
**Pull request by pgomulka**
This commit adds an example stable-analysis plugin under plugins/examples to show a usage of the new stable analysis api.
It implements AnalyzerFactory, CharFilterFactory, TokenFilterFactory and TokenizerFactory. 
it also presents how to write unit tests and yaml rest tests without the dependencies on elasticsearch test framework.

relates https://github.com/elastic/elasticsearch/issues/88980

### CODE REVIEW

File: plugins/examples/stable-analysis/build.gradle [hunk 1]
```diff
@@ -0,0 +1,34 @@
+//TODO guide - mention buildscript (in parent)
```
Comment (williamrandolph) [inline plugins/examples/stable-analysis/build.gradle:1]: Note: "buildscript" is a code block in `build.gradle`

**Comment by pgomulka** [comment]
@elasticmachine  ok to test

**Comment by williamrandolph** [comment]
@elasticmachine test this please

**Review by williamrandolph — APPROVED** [review]
This is great work. I've run a successful live demo with it, and will be creating a guide for it soon.

**Pull request merged**